src/backend/utils/gdd/gdddetectorpriv.h

```diff
@@ -46,9 +43,8 @@
  */
 #define gdd_list_iter_delete(iter) do \
 { \
-	(iter).list = list_delete_cell((iter).list, (iter).cell, (iter).prev); \
-	(iter).prev = (iter).list != NIL ? (iter).prev : NULL; \
-	(iter).cell = (iter).prev; \
+	(iter).list = list_delete_cell((iter).list, (iter).cell); \
+	(iter).cell = (iter).list != NIL ? (iter).cell : NULL; \
```

Comment:
It seems to me that here we also have a problem with skipping an element after deletion. Since after deleting the elements, they will shift and the current element will become the next one, after which we will move on to the next one again.

Reply:
fixed